### PR TITLE
Remove `async_trait` macro in favor of AFIT & RPITIT

### PR DESCRIPTION
**Pull request by thomaseizinger**
Resolves: #202.

### CODE REVIEW

File: xtra/src/lib.rs [hunk 1]
```diff
@@ -114,16 +113,16 @@ use crate::recv_future::Message;
...
     /// Handle a given message, returning its result.
-    ///
-    /// This is an [`async_trait`](https://docs.rs/async-trait).
-    /// See the trait documentation to see an example of how this method can be declared.
-    async fn handle(&mut self, message: M, ctx: &mut Context<Self>) -> Self::Return;
+    fn handle(
+        &mut self,
+        message: M,
+        ctx: &mut Context<Self>,
+    ) -> impl Future<Output = Self::Return> + Send;
```
Comment (thomaseizinger) [inline xtra/src/lib.rs:125]: A limitation of the current feature is that there is no "return-type notation", meaning users of this trait cannot specify, whether or not this future is `Send`. If we would just use `async fn` here in the declaration, this `Future` would not be `Send` and thus we would not be able to use `xtra` on multi-threaded executors.

From reading https://github.com/rust-lang/rust/pull/115822, this is what is most likely going to be stablized. It is still quite cool that this just works now! :)

Comment (thomaseizinger) [reply xtra/src/lib.rs:125]: As we can see in the rest of the codebase, implementations can just use `async fn` and they will inherit the `Send` requirement.

**Review by Restioson — COMMENTED** [review]
Very cool! I'm surprised that MessageEnvelope didn't need any changes

**Comment by thomaseizinger** [comment]
https://github.com/rust-lang/rust/pull/115822 just merged which will be stable in Rust 1.75 on December 28th! :partying_face:

**Comment by Restioson** [comment]
Woo! Awesome!

**Comment by thomaseizinger** [comment]
@Restioson I think this is now ready for being merged once `1.75` is released.

**Comment by Restioson** [comment]
Awesome! :tada:

**Comment by thomaseizinger** [comment]
@Restioson This is now ready to be merged!

**Review by Restioson — APPROVED** [review]

**Comment by thomaseizinger** [comment]
Shall we release `0.6` finally? :partying_face:

**Pull request merged**